site/src/components/UpdateCheckBanner/UpdateCheckBanner.tsx

```diff
@@ -0,0 +1,48 @@
+import Link from "@material-ui/core/Link"
+import { AlertBanner } from "components/AlertBanner/AlertBanner"
+import { Trans, useTranslation } from "react-i18next"
+import * as TypesGen from "api/typesGenerated"
+
+export interface UpdateCheckBannerProps {
+  updateCheck?: TypesGen.UpdateCheckResponse
+  error?: Error | unknown
+  onDismiss?: () => void
+}
+
+export const UpdateCheckBanner: React.FC<
+  React.PropsWithChildren<UpdateCheckBannerProps>
+> = ({ updateCheck, error, onDismiss }) => {
+  const { t } = useTranslation("common")
+
+  return (
+    <>
+      {!error && updateCheck && !updateCheck.current && (
+        <AlertBanner severity="info" onDismiss={onDismiss} dismissible>
+          <div>
+            <Trans
+              t={t}
+              i18nKey="updateCheck.message"
+              values={{ version: updateCheck.version }}
+            >
+              Coder {"{{version}}"} is now available. View the{" "}
+              <Link href={updateCheck.url}>release notes</Link> and{" "}
+              <Link href="https://coder.com/docs/coder-oss/latest/admin/upgrade">
+                upgrade instructions
+              </Link>{" "}
+              for more information.
+            </Trans>
+          </div>
+        </AlertBanner>
+      )}
+      {error && (
+        <AlertBanner
+          severity="error"
+          error={error}
+          text={t("updateCheck.error")}
+          onDismiss={onDismiss}
+          dismissible
+        />
+      )}
+    </>
```

Comment:
Just curious why we throw an error if the check fails. Perhaps it would be better if this failed silently (we could throw something in the terminal if we wanted)? Maybe not, just a thought after seeing the error on local:
![Screen Shot 2022-11-30 at 5 02 28 PM](https://user-images.githubusercontent.com/19142439/204917713-a82500cb-cec9-458f-9cad-4a082711d7e7.png)

Reply:
I thought about it and I worried we wouldn't notice if it stops working for whatever reason. I can make it silent or add a prefix (`Update check failed: Route not found.`). Ideally we would keep it silent and log it to Sentry or a similar bug tracker.

Reply:
Yeah, maybe we could just add a console.error since we don't have Sentry.

Reply:
I refactored this now, it looks like this: https://www.chromatic.com/test?appId=624de63c6aacee003aa84340&id=6388bfa22e0a55f2ff0c1f1e

I'm still game to remove it if that's preferred. 👍🏻

Reply:
@mafredri nope looks good!